src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useContext } from 'react';
+import { Filter } from '../Filter/Filter';
+import { TodoContext } from '../Store/TodoContext';
+import { deleteTodos } from '../../api/todos';
+
+export const Footer: React.FC = () => {
+  const { todos, setTodos, setErrorMessage } = useContext(TodoContext);
+
+  const hasEnoughTodos = todos.length > 0;
+
+  const uncompletedTodos = todos.filter(todo => !todo.completed);
+
+  const hasEnoughCompletedTodo = todos.some(todo => todo.completed);
+
+  const todoInput = document.getElementById('todoInput');
+
+  const handleDelete = (todoId: number) => {
+    deleteTodos(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(post => post.id !== todoId),
+        );
+      })
+      .catch(error => {
+        setErrorMessage('Unable to delete a todo');
+        setTodos(todos);
+        throw error;
```

Comment:
```suggestion
```
it is redundant. setErrorMessage is enough for handling error. and we need to move on finally block.